torch/distributions/distribution.py

```diff
@@ -10,10 +10,17 @@ class Distribution(object):
 
     has_rsample = False
     has_enumerate_support = False
+    support = None
```

Comment:
I think it's cleaner to make these `@property`s that `raise NotImplementedError`

Reply:
Sure. In fact, I just noticed that these are already added as `property`.